application/config/config.go

```diff
@@ -253,7 +253,11 @@ func initWorkFlowEngine(c *Config) {
 	conf.Set(configuration.FF_OAUTH_AUTH_FLOW_ENABLED, enableOAuth)
 
 	c.engine = app.CreateAppEngineWithOptions(app.WithConfiguration(conf), app.WithZeroLogger(c.logger))
-	c.storage = storage.NewStorage()
+	var storageCreationErr error
+	c.storage, storageCreationErr = storage.NewStorageWithCallbacks()
+	if storageCreationErr != nil {
+		c.logger.Err(storageCreationErr).Msg("unable to create a configuration storage object")
+	}
 	conf.SetStorage(c.storage)
```

Comment:
This raised an eyebrow (wondered about a possible nil deref?) but looking at the `extendedViper` configuration.Configuration implementation in the GAF, I think this should be fine.

A comment that we're assuming the config implementation will tolerate this might be a good idea (the GAF might change in the future, and the toleration along with it).